src/library/scala/util/Either.scala

```diff
@@ -54,44 +54,114 @@ package util
  *  val left42 = Left(42.0)
  *
  *  for {
- *    a <- right1
- *    b <- right2
- *    c <- right3
- *  } yield a + b + c // Right(6)
+ *    x <- right1
+ *    y <- right2
+ *    z <- right3
+ *  } yield x + y + z // Right(6)
  *
  *  for {
- *    a <- right1
- *    b <- right2
- *    c <- left23
- *  } yield a + b + c // Left(23.0)
+ *    x <- right1
+ *    y <- right2
+ *    z <- left23
+ *  } yield x + y + z // Left(23.0)
  *
  *  for {
- *    a <- right1
- *    b <- left23
- *    c <- right2
- *  } yield a + b + c // Left(23.0)
+ *    x <- right1
+ *    y <- left23
+ *    z <- right2
+ *  } yield x + y + z // Left(23.0)
  *
- *  // It may be necessary to provide the type of the “missing” value, especially the type
- *  // of the right value for `Left`. Otherwise, without any context that constrains the type,
- *  // it might be inferred as `Nothing`:
+ *  // Guard expressions are not supported:
  *  for {
- *    a <- left23
- *    b <- right1
- *    c <- left42  // type at this position: Either[Double, Nothing]
- *  } yield a + b + c
+ *    i <- right1
+ *    if i > 0
+ *  } yield i
+ *  // error: value withFilter is not a member of Right[Double,Int]
+ *
+ *  // Similarly, refutable patterns are not supported:
+ *  for (x: Int <- right1) yield x
+ *  // error: value withFilter is not a member of Right[Double,Int]
+ *  }}}
+ *
+ *  Since `for` comprehensions use `map` and `flatMap`, the types
+ *  of function parameters used in the expression must be inferred.
+ *  These types are constrained by the `Either` values. In particular,
+ *  because of right-biasing, `Left` values may require an explicit
+ *  type argument for type parameter `B`, the right value. Otherwise,
+ *  it might be inferred as `Nothing`.
+ *
+ *  {{{
+ *  for {
+ *    x <- left23
+ *    y <- right1
+ *    z <- left42  // type at this position: Either[Double, Nothing]
+ *  } yield x + y + z
  *  //            ^
  *  // error: ambiguous reference to overloaded definition,
  *  // both method + in class Int of type (x: Char)Int
  *  // and  method + in class Int of type (x: Byte)Int
  *  // match argument types (Nothing)
+ *
+ *  for (x <- right2 ; y <- left23) yield x + y  // Left(23.0)
+ *  for (x <- right2 ; y <- left42) yield x + y  // error
+ *
+ *  for {
+ *    x <- right1
+ *    y <- left42  // type at this position: Either[Double, Nothing]
+ *    z <- left23
+ *  } yield x + y + z
+ *  // Left(42.0), but unexpectedly a `Either[Double,String]`
  *  }}}
  *
  *  @author <a href="mailto:research@workingmouse.com">Tony Morris</a>, Workingmouse
  *  @version 2.0, 2016-07-15
  *  @since 2.7
  */
 sealed abstract class Either[+A, +B] extends Product with Serializable {
-  /** Projects this `Either` as a `Left`. */
+  /** Projects this `Either` as a `Left`.
```

Comment:
Moved this doc from LeftProjection to `left`, where it's more natural to click for it.